dace/transformation/interstate/state_fusion.py

```diff
@@ -132,8 +132,14 @@ def memlets_intersect(graph_a: SDFGState, group_a: List[nodes.AccessNode],
 
     @staticmethod
     def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
-        first_state = graph.nodes()[candidate[StateFusion.first_state]]
-        second_state = graph.nodes()[candidate[StateFusion.second_state]]
+        if isinstance(candidate[StateFusion.first_state], SDFGState):
+            first_state: SDFGState = candidate[StateFusion.first_state]
+            second_state: SDFGState = candidate[StateFusion.second_state]
+        else:
+            first_state: SDFGState = graph.node(
+                candidate[StateFusion.first_state])
+            second_state: SDFGState = graph.node(
+                candidate[StateFusion.second_state])
```

Comment:
What's going on here? Looks like it's working around two conflicting conventions?

Reply:
Exactly. I now added a comment and it will be removed in the PR that changes the Transformation API once and for all.